provider/gce/environ_network.go

```diff
@@ -136,16 +138,21 @@ func (e *environ) getInstanceSubnets(
 }
 
 // NetworkInterfaces implements environs.NetworkingEnviron.
-func (e *environ) NetworkInterfaces(ctx context.ProviderCallContext, instId instance.Id) ([]network.InterfaceInfo, error) {
-	insts, err := e.Instances(ctx, []instance.Id{instId})
-	if err != nil {
-		return nil, errors.Trace(err)
+func (e *environ) NetworkInterfaces(ctx context.ProviderCallContext, ids []instance.Id) ([][]network.InterfaceInfo, error) {
+	if len(ids) == 0 {
+		return nil, environs.ErrNoInstances
 	}
-	envInst, ok := insts[0].(*environInstance)
-	if !ok {
-		// This shouldn't happen.
-		return nil, errors.Errorf("couldn't extract google instance for %q", instId)
+
+	// Fetch instance information for the IDs we are interested in.
+	insts, err := e.Instances(ctx, ids)
+	partialInfo := err == environs.ErrPartialInstances
+	if err != nil && err != environs.ErrPartialInstances {
+		if err == environs.ErrNoInstances {
```

Comment:
We really should be looking at `errors.Cause` rather than test equality where we care about `ErrNoInstances`.
Perhaps add a TODO?

Reply:
Good point. I will push a commit to change this